### ISSUE TITLE
Notifications stay on the screen and error appears - Cannot read property 'height' of undefined

### ISSUE THREAD

**Issue by erapoport** [lumapps/lumX#366]
Tested with: chrome, firefox

I think it mostly happens when there are multiple notifications at the same time. It happens with the notifications in the demo as well, just click enough to fill half of the screen and then notifications start throwing that error and hanging on the screen.

**Comment by steveoc64** [comment]
Duplicate behavior here.

I can get the same issue with 2 notifications on the screen.   Will have a look at debugging it when I get some more time.

**Comment by dandanknight** [comment]
I was having this issue also. Did some investigation and found that it's down to the 400ms $timeout in the deleteNotification function. I think they put the timeout there to wait for a digest cycle to happen, but the delay was causing some race condition type thing where it was trying to find something in an array that had already been removed by a previous splice.
If you remove the 400 but keep the $timeout, it seems to solve the issue, as a $timeout without a delay specified is basically a safe was of doing a $scope.$apply without getting the dreaded "digest cycle in progress" message.
In the delete notification function, if you change this....
````
            $timeout(function () {
                notification.elem.remove();
                notificationList.splice(notifIndex, 1);
            }, 400);
```` 
to this....
````
            $timeout(function () {
                notification.elem.remove();
                notificationList.splice(notifIndex, 1);
            });
```` 
it fixes the issue. If I have time, I'll create a PR.

Cheers,

Dan

**Comment by matmkian** [comment]
Fixed on v1.0.4 :-)
